### ISSUE TITLE
TinyMCE does not save HTML code properly

### ISSUE THREAD

**Issue by AndyReiter** [craftercms/craftercms#7079]
### CI Number

5231

### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] The issue is in the develop branch
- [X] The issue is in the latest released 4.1.x
- [ ] The issue is in the latest released 4.0.x
- [ ] The issue is in the latest released 3.1.x

### Describe the issue

When you enter valid HTML code into the RTE Code Sample, all the HTML code is incorrectly saved and rendered. See video for reproduction.

### Steps to reproduce

Steps:

1. Create a Site based on Editorial
2. Edit the home page
3. Navigate to the RTE editor
4. Insert a new "Code Sample" and add the following HTML code `<p><em>Testing</em></p>`
5. Save & Close
6. Notice the UI doesn't render the HTML code properly, and displays the `<p>` and `<em>` tags
7. Now edit the home page again
8. Select Tools --> Code Editor
9. Modify the code to now be `<p><em>Testing again</em></p>` and save&close
10. Notice the UI renders the HTML code properly

### Relevant log output

```shell
N/A
```

### Screenshots and/or videos

https://www.loom.com/share/2f66549515f9427db9e1b4ee7900fb49?sid=5265a726-9e68-4c8d-8437-c4a61288afdb

**Comment by rart** [comment]
This is due to the editor cleaning up code. To allow, due to the way the code samples plugin writes the code, a change in configuration is required:
```json
{
  ...
  "extended_valid_elements": "pre[class|name|id|style],code[class|name|id|style]",
  "valid_children": "+pre[code]",
  ...
}
```

**Comment by sumerjabri** [comment]
@alhambrav please add to release notes.

**Comment by sumerjabri** [comment]
We must fix all BPs

**Comment by sumerjabri** [comment]
@alhambrav please document how to edit codesamples in the authoring guide double-clicking the code sample, or using the same menu option used to create it. @rart can explain further.

Perhaps as a note here: https://docs.craftercms.org/en/4.1/by-role/author/index.html#working-in-the-rich-text-editor-rte

**Comment by sumerjabri** [comment]
@russdanner, the code review of the docs is with you.

**Comment by AndyReiter** [comment]
Confirmed fixed in latest build of 4.2 develop. This can be closed.
